packages/hash/frontend/src/components/hooks/blockProtocolFunctions/useBlockProtocolCreateLinks.ts

```diff
@@ -45,66 +33,35 @@ export const useBlockProtocolCreateLinks = (): {
           throw new Error("createLinks needs to be passed a sourceAccountId");
         }
 
-        const baseFields = {
-          path: action.path,
-          sourceEntityId: action.sourceEntityId,
-          sourceAccountId: action.sourceAccountId,
-        };
-
-        if ("operation" in action) {
-          if (!action.operation.entityTypeId) {
-            throw new Error(
-              "entityTypeId is compulsory on operation while trying to create a linkedAggregation",
-            );
-          }
-
-          const { data, errors } = await runCreateLinkedAggregationMutation({
-            variables: {
-              ...action,
-              operation: {
-                // @todo this shouldn't be necessary
-                ...action.operation,
-                entityTypeId: action.operation.entityTypeId,
-              },
-              sourceAccountId: action.sourceAccountId,
-            },
-          });
-          if (!data) {
-            throw new Error(`Could not create link: ${errors?.[0]!.message}`);
-          }
-
-          // @todo, add a proper typecheck. The GraphQL query for multiFilter { operator } returns String, but BlockProtocolLinkedAggregationUpdateMutationResults defines the exact type for operator. This typecast is used to typecast string to the one the query expects.
-          results.push(
-            data.createLinkedAggregation as unknown as BlockProtocolLink,
-          );
-        } else if (!("destinationAccountId" in action)) {
+        if (!("destinationAccountId" in action)) {
           throw new Error(
             "One of operation or destinationEntityId must be provided",
           );
-        } else {
-          const newLink = {
-            ...baseFields,
-            destinationEntityId: action.destinationEntityId,
-            destinationAccountId: action.destinationAccountId
-              ? action.destinationAccountId
-              : action.sourceAccountId,
-          };
+        }
```

Comment:
**BLOCKING** (for incorrect error message): don't think we need this - `destinationAccountId` is not needed, because the source's accountId will be used as a fallback below.

If we want to keep this required, we should remove the fallback (as it'll never be reached), and update the error message (since it mentions 'operation', which will never appear here, and 'destinationEntityId', which isn't the property being checked for)

Reply:
Good spot, sorry this was leftover from the prior version of the `useBlockProtocolCreateLinks` which I believe intended to check for `destinationEntityId` (instead of `destinationAccountId`) in the action.

Since the types no longer allow this anyway, I've removed this check entirely in https://github.com/hashintel/hash/pull/497/commits/775ceb7d7648b3b048cffa74ec47d382dcebd56f